From 826cad26e9dfcfd69162dba055290062fb026ac4 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 11 Mar 2024 11:39:47 +0100
Subject: [PATCH 539/954] fix(stm32mp2): keep ECSDA key and signature
 definitions

The patch [1] was removing BOOT_API_ECDSA_PUB_KEY_LEN_IN_BYTES and
BOOT_API_ECDSA_SIGNATURE_LEN_IN_BYTES, and defining 256 and 384
versions of the macros. But the removed macros are required in
crypto lib when compiling with trusted boot. Redefine them using
the 256/384 version depending on SoC is STM32MP1 or not.

[1] 4cf19ad4de feat(stm32mp21): update boot API header version

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git fa50f8b61a27880c29fa45e65088b3666ec4968b fix(stm32mp2): keep ECSDA key and signature definitions]
Change-Id: I4ec8e8b9d249ed542dda9bd64ef6248f8a56ef62
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/365744
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 plat/st/stm32mp2/include/boot_api.h | 25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/plat/st/stm32mp2/include/boot_api.h b/plat/st/stm32mp2/include/boot_api.h
index 79dbe8d17a..53e0c72c9a 100644
--- a/plat/st/stm32mp2/include/boot_api.h
+++ b/plat/st/stm32mp2/include/boot_api.h
@@ -108,6 +108,18 @@
 #define BOOT_API_ECDSA_SIGNATURE_384_LEN_IN_BYTES		96
 #define BOOT_API_SHA384_DIGEST_SIZE_IN_BYTES			48
 
+#if STM32MP21
+#define BOOT_API_ECDSA_PUB_KEY_LEN_IN_BYTES			\
+	BOOT_API_ECDSA_PUB_KEY_384_LEN_IN_BYTES
+#define BOOT_API_ECDSA_SIGNATURE_LEN_IN_BYTES			\
+	BOOT_API_ECDSA_SIGNATURE_384_LEN_IN_BYTES
+#else /* !STM32MP21 */
+#define BOOT_API_ECDSA_PUB_KEY_LEN_IN_BYTES			\
+	BOOT_API_ECDSA_PUB_KEY_256_LEN_IN_BYTES
+#define BOOT_API_ECDSA_SIGNATURE_LEN_IN_BYTES			\
+	BOOT_API_ECDSA_SIGNATURE_256_LEN_IN_BYTES
+#endif /* STM32MP21 */
+
 /* Possible values of the field 'boot_api_image_header_t.ecc_algo_type' */
 #define BOOT_API_ECDSA_ALGO_TYPE_P256NIST			1
 #define BOOT_API_ECDSA_ALGO_TYPE_BRAINPOOL256			2
@@ -323,11 +335,7 @@ typedef struct {
 typedef struct {
 	/* BOOT_API_IMAGE_HEADER_MAGIC_NB */
 	uint32_t magic;
-#if STM32MP21
-	uint8_t image_signature[BOOT_API_ECDSA_SIGNATURE_384_LEN_IN_BYTES];
-#else /* STM32MP21 */
-	uint8_t image_signature[BOOT_API_ECDSA_SIGNATURE_256_LEN_IN_BYTES];
-#endif /* STM32MP21 */
+	uint8_t image_signature[BOOT_API_ECDSA_SIGNATURE_LEN_IN_BYTES];
 	/*
 	 * Checksum of payload
 	 * 32-bit sum all payload bytes considered as 8 bit unsigned
@@ -408,12 +416,7 @@ typedef struct {
 	 */
 	uint32_t ecc_algo_type;
 	/* ECDSA public key to be used to check signature. */
-#if STM32MP21
-	uint8_t ecc_pubk[BOOT_API_ECDSA_PUB_KEY_384_LEN_IN_BYTES];
-#else /* STM32MP21 */
-	uint8_t ecc_pubk[BOOT_API_ECDSA_PUB_KEY_256_LEN_IN_BYTES];
-#endif /* STM32MP21 */
-
+	uint8_t ecc_pubk[BOOT_API_ECDSA_PUB_KEY_LEN_IN_BYTES];
 	/* table of Hash of Algo+ECDSA public key */
 #if STM32MP21
 	boot_api_sha384_t pk_hashes[];
-- 
2.53.0

